install_files/ansible-base/roles/restore/tasks/main.yml

```diff
@@ -42,6 +43,16 @@
     state: absent
   when: torrc_check_dir.path is defined
 
+- name: Verify that the backup Tor config is compatible with the server Tor config
+  assert:
+    that:
+      - "'Valid configuration' in compare_result.stdout"
+    fail_msg:
+      - "This backup's tor configuration cannot be applied on this server."
+      - "A data-only restore can be applied using the --preserve-tor-config argument"
+      - "More info: {{ compare_result.stdout }}"
+  when: restore_skip_tor is not defined
```

Comment:
Nit: consider setting `restore_skip_tor` to False in `defaults/main.yml`, then update this line to be `when: not restore_skip-tor`. Checking for definedness works in this case, but only because it's non-existent, or false.